File: .github/skills/python-mcp-development/templates/basic-server.py
#!/usr/bin/env python3
"""
基本的なMCPサーバーテンプレート
"""
from mcp.server.fastmcp import FastMCP
from pydantic import BaseModel, Field

# FastMCPサーバーの初期化
mcp = FastMCP("Basic MCP Server")


# --- ツールの定義 ---

@mcp.tool()
def echo(message: str) -> str:
    """メッセージをエコーバックします。
    
    Args:
        message: エコーするメッセージ
    
    Returns:
        入力されたメッセージ
    """
    return message


class CalculationResult(BaseModel):
    """計算結果の構造化データ"""
    operation: str = Field(description="実行された演算")
    operand1: float = Field(description="第一オペランド")
    operand2: float = Field(description="第二オペランド")
    result: float = Field(description="計算結果")


@mcp.tool()
def calculate(a: float, b: float, operation: str) -> CalculationResult:
    """2つの数値で四則演算を実行します。
    
    Args:
        a: 第一オペランド
        b: 第二オペランド
        operation: 演算子 (add, subtract, multiply, divide)
    
    Returns:
        計算結果を含む構造化データ
    """
    if operation == "add":
        result = a + b
    elif operation == "subtract":
        result = a - b
    elif operation == "multiply":
        result = a * b
    elif operation == "divide":
        if b == 0:
            raise ValueError("ゼロ除算エラー")
        result = a / b
    else:
        raise ValueError(f"未知の演算: {operation}")
    
    return CalculationResult(
        operation=operation,
        operand1=a,
        operand2=b,
        result=result
    )


# --- リソースの定義 ---

@mcp.resource("info://server")
def get_server_info() -> str:
    """サーバー情報を提供する静的リソース"""
    return """
    Server: Basic MCP Server
    Version: 0.1.0
    Capabilities: tools, resources
    """


# --- サーバー起動 ---

if __name__ == "__main__":
    # STDIOトランスポートでサーバーを起動
    mcp.run()

File: .claude/skills/typescript-mcp-development/templates/basic-server.ts
#!/usr/bin/env node
/**
 * 基本的なMCPサーバーテンプレート
 */
import { McpServer } from "@modelcontextprotocol/sdk/server/mcp.js";
import { StdioServerTransport } from "@modelcontextprotocol/sdk/server/stdio.js";
import { z } from "zod";

// サーバーの初期化
const server = new McpServer({
  name: "basic-mcp-server",
  version: "1.0.0",
});

// --- ツールの定義 ---

server.registerTool({
  name: "echo",
  title: "Echo Message",
  description: "Echo back the provided message",
  inputSchema: {
    message: z.string().describe("Message to echo"),
  },
}, async ({ message }) => {
  return {
    content: [{ type: "text", text: message }],
  };
});

interface CalculationResult {
  operation: string;
  operand1: number;
  operand2: number;
  result: number;
}

server.registerTool({
  name: "calculate",
  title: "Calculate Numbers",
  description: "Perform basic arithmetic operations",
  inputSchema: {
    a: z.number().describe("First operand"),
    b: z.number().describe("Second operand"),
    operation: z.enum(["add", "subtract", "multiply", "divide"])
      .describe("Operation to perform"),
  },
}, async ({ a, b, operation }) => {
  let result: number;

  switch (operation) {
    case "add":
      result = a + b;
      break;
    case "subtract":
      result = a - b;
      break;
    case "multiply":
      result = a * b;
      break;
    case "divide":
      if (b === 0) {
        throw new Error("Division by zero");
      }
      result = a / b;
      break;
  }

  const structured: CalculationResult = {
    operation,
    operand1: a,
    operand2: b,
    result,
  };

  return {
    content: [{ type: "text", text: `Result: ${result}` }],
    structuredContent: structured,
  };
});

// --- リソースの定義 ---

server.registerResource({
  name: "server_info",
  uri: "info://server",
  title: "Server Information",
  description: "Get server information",
}, async () => {
  return {
    contents: [{
      uri: "info://server",
      mimeType: "application/json",
      text: JSON.stringify({
        name: "basic-mcp-server",
        version: "1.0.0",
        capabilities: ["tools", "resources"],
      }, null, 2),
    }],
  };
});

// --- サーバー起動 ---

async function main() {
  const transport = new StdioServerTransport();
  await server.connect(transport);

  // 注意: console.log()は使用しない（stdoutを汚染する）
  console.error("Basic MCP Server running on stdio");
}

main().catch((error) => {
  console.error("Server error:", error);
  process.exit(1);
});
